sys/dev/ath/if_ath_usb.c

```diff
@@ -1673,7 +1675,9 @@ ath_usb_write(struct ath_softc *sc, uint32_t addr, uint32_t val)
 
 	usc->wbuf[usc->wcount].addr = htobe32(addr);
 	usc->wbuf[usc->wcount].val  = htobe32(val);
-	if (++usc->wcount == AR_MAX_WRITE_COUNT)
+	if (!usc->no_buffer_write & (++usc->wcount == AR_MAX_WRITE_COUNT))
```

Comment:
Shouldn't it be more like (without `else`):
```C
	if ((++usc->wcount == AR_MAX_WRITE_COUNT) || (usc->no_buffer_write))
		ath_usb_write_barrier(sc);
```

Reply:
I don't think so because then we inceremnt usc->wcount when no_buffer_write is true

Reply:
I changed as you suggested.